### PR TITLE
[SHELL32] Add missing desktop.ini files

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
Add the "desktop.ini" files to the special folders to realize the custom icons of them.
JIRA issue: [CORE-10045](https://jira.reactos.org/browse/CORE-10045)

### CODE REVIEW

**Comment by katahiromz** [comment]
AFTER:
folder-icons-after

**Comment by learn-more** [comment]
This file is (partially?) wine-synced:
https://source.winehq.org/git/wine.git/blob/85f1a79be8bbb9b62ad88e0d0fbf1e005fd3ba16:/dlls/shell32/shellpath.c

**Comment by katahiromz** [comment]
Can someone coordinate this for Wine? I don't like Wine.

**Comment by Doug-Lyons** [comment]
The latest README.WINE has this line:
reactos/dll/win32/shell32             # Forked at Wine-20071011

File: dll/win32/shell32/wine/shellpath.c [hunk 1]
```diff
@@ -2128,8 +2158,46 @@ HRESULT WINAPI SHGetFolderPathAndSubDirW(
...
+        /* build the desktop.ini file path */
+        PathAppendW(szBuildPath, s_szDesktopIni);
+
+        /* build the icon location */
+        StringCchPrintfW(szIconLocation, _countof(szIconLocation), s_szFormat,
+                         CSIDL_Data[folder].nShell32IconIndex);
+
+        /* write desktop.ini */
+        WritePrivateProfileStringW(s_shellClassInfo, s_szIconResource, szIconLocation, szBuildPath);
+        WritePrivateProfileStringW(NULL, NULL, NULL, szBuildPath);
```
Comment (HBelusca) [inline dll/win32/shell32/wine/shellpath.c]: What does this function do when the three first parameters are NULL ?!

File: dll/win32/shell32/wine/shellpath.c [hunk 2]
```diff
@@ -2128,8 +2158,46 @@ HRESULT WINAPI SHGetFolderPathAndSubDirW(
         hr = E_FAIL;
         goto end;
     }
-
     TRACE("Created missing system directory %s\n", debugstr_w(szBuildPath));
```
Comment (HBelusca) [inline dll/win32/shell32/wine/shellpath.c]: May move that trace at the very end (just before the "end:" label) ?

File: dll/win32/shell32/wine/shellpath.c (hunk 1 above)
Comment (katahiromz) [reply dll/win32/shell32/wine/shellpath.c]: https://docs.microsoft.com/en-us/windows/desktop/api/winbase/nf-winbase-writeprivateprofilestringa

> After following the preceding steps, an application setup program should call WritePrivateProfileString with the first three parameters set to NULL, and the fourth parameter set to the INI file name. For example: WritePrivateProfileString( NULL, NULL, NULL, L"appname.ini" );
> Such a call causes the mapping of an .ini file to the registry to take effect before the next system reboot. The system rereads the mapping information into shared memory. A user will not have to reboot their computer after installing an application in order to have future invocations of the application see the mapping of the .ini file to the registry.

File: dll/win32/shell32/wine/shellpath.c (hunk 2 above)
Comment (katahiromz) [reply dll/win32/shell32/wine/shellpath.c]: Excuse me?

File: dll/win32/shell32/wine/shellpath.c (hunk 1 above)
Comment (katahiromz) [reply dll/win32/shell32/wine/shellpath.c]: I added a simple comment. OK.

File: dll/win32/shell32/wine/shellpath.c (hunk 2 above)
Comment (katahiromz) [reply dll/win32/shell32/wine/shellpath.c]: Don't move. OK.

**Comment by katahiromz** [comment]
Done.

**Pull request merged**